src/ts/replayprotection/replayprotectionauthority.ts

```
@@ -37,20 +37,19 @@ export abstract class ReplayProtectionAuthority {
    * @param index Index in Nonce Store
    * @param contract Hub Contract
    */
-  protected async accessNonceStore(
-    signer: Wallet,
-    index: BigNumber,
-    contract: string
-  ): Promise<BigNumber> {
+  protected async accessNonceStore(index: BigNumber): Promise<BigNumber> {
     try {
-      const replayProtection = new ReplayProtectionFactory(signer).attach(
-        contract
+      const replayProtection = new ReplayProtectionFactory(this.signer).attach(
+        this.forwarderAddress
       );
       // In the ReplayProtection.sol, we use latestNonce == storedNonce then continue.
       // Onchain ID = H(signerAddress, index).
       // Mostly benefits bitflip & multinonce.
       const onchainId = keccak256(
-        defaultAbiCoder.encode(["address", "uint"], [signer.address, index])
+        defaultAbiCoder.encode(
+          ["address", "uint"],
+          [this.signer.address, index]
+        )
       );
 
       return await replayProtection.nonceStore(onchainId);
```

You;re returning BigNumber("0") on the line below, and making an assumption about the error.

You should correctly classify the error, tbh, I would suggest not catching at all and allowing it to bubble up. At the moment if an error is thrown and caught here because the contract isnt deployed, it will return 0, at which point the upper layer will consider this to be a valid nonce and return it, when really they still need to deploy the contract.

---

so the idea here is that; if the proxy account does not yet exist then the library is still usable. 

so the user will send up two meta-tx, one for deploy, one for the command. and the library knows how to deal with that.

---

If you want to return a nonce for the case where a contract hasn't been deployed, you should check explicitly whether the contract has been deployed after receiving the error. At the moment you're swallowing all possible errors.

---

i just used the code you mentioned earlier to see if code exists. so it'll return 0 if it doesnt exist, otherwise it'll try to access the nonce store.